### PR TITLE
Extract launcher from main composable

### PR DESCRIPTION
**Pull request by gmatyszczak**

### CODE REVIEW

File: app/src/main/kotlin/com/intive/picover/main/navigation/launcher/Launcher.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.intive.picover.main.navigation.launcher
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+
+@Singleton
+class Launcher @Inject constructor() {
```
Comment (gmatyszczak) [inline app/src/main/kotlin/com/intive/picover/main/navigation/launcher/Launcher.kt:10]: The idea is to inject that later into ios view controller, in the swift code, and launch there sign in screen from firebase auth UI for swift. In theory should work 😄  We should be able to use this https://skie.touchlab.co/features/flows to observe flow in swift

Comment (MarekMacko) [reply app/src/main/kotlin/com/intive/picover/main/navigation/launcher/Launcher.kt:10]: Sounds promising 😄

File: app/src/main/kotlin/com/intive/picover/main/viewmodel/MainViewModel.kt [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
 import javax.inject.Inject
 import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
 
-@HiltViewModel
 class MainViewModel @Inject constructor(
 	authRepository: AuthRepository,
-) : ViewModel() {
+	launcher: Launcher,
+	val snackbarHostState: SnackbarHostState,
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/main/viewmodel/MainViewModel.kt:16]: Is it needed?

File: app/src/test/kotlin/com/intive/picover/main/viewmodel/MainViewModelTest.kt [hunk 3]
```diff
@@ -15,7 +18,8 @@ import kotlinx.coroutines.flow.flowOf
 class MainViewModelTest : ShouldSpec(
 	{
 		val authRepository: AuthRepository = mockk()
-		val tested by lazy { MainViewModel(authRepository) }
+		val launcher: Launcher = mockk(relaxUnitFun = true)
+		fun tested() = MainViewModel(authRepository, launcher, mockk())
```
Comment (MarekMacko) [inline app/src/test/kotlin/com/intive/picover/main/viewmodel/MainViewModelTest.kt]: Wouldn't be better to use property, as before, with custom getter?

Comment (gmatyszczak) [reply app/src/test/kotlin/com/intive/picover/main/viewmodel/MainViewModelTest.kt]: That was also my initial idea, but it's available only on class properties, and here we are in scope of lambda from ShouldSpec, which has some downsides like this one 😄

File: app/src/main/kotlin/com/intive/picover/main/viewmodel/MainViewModel.kt (hunk 2 above)
Comment (gmatyszczak) [reply app/src/main/kotlin/com/intive/picover/main/viewmodel/MainViewModel.kt:16]: It's accessed in MainScreen to set up snackbar host

Comment (MarekMacko) [reply app/src/main/kotlin/com/intive/picover/main/viewmodel/MainViewModel.kt:16]: Right, now I see

File: app/src/test/kotlin/com/intive/picover/main/viewmodel/MainViewModelTest.kt (hunk 3 above)
Comment (MarekMacko) [reply app/src/test/kotlin/com/intive/picover/main/viewmodel/MainViewModelTest.kt]: Good to know 😄  Maybe we can use `isolationMode = IsolationMode.InstancePerTest` and keep `tested` as previously?

**Review by MarekMacko — APPROVED** [review]

File: app/src/test/kotlin/com/intive/picover/main/viewmodel/MainViewModelTest.kt (hunk 3 above)
Comment (gmatyszczak) [reply app/src/test/kotlin/com/intive/picover/main/viewmodel/MainViewModelTest.kt]: That should work, thanks for the hint! 😄

**Pull request merged**